app/jobs/split_pdf_job.rb

```diff
@@ -35,14 +36,46 @@ def perform(exam_template, path)
         m = qrcode_regex.match qrcode_string
         if m.nil?
           new_page.save File.join(error_dir, "#{basename}-#{i}.pdf")
+          num_pages_qr_scan_error += 1
         else
           partial_exams[m[:exam_num]] << [m[:page_num].to_i, page]
           m_logger.log("#{m[:short_id]}: exam number #{m[:exam_num]}, page #{m[:page_num]}")
         end
       end
-
       save_pages(exam_template, partial_exams)
+
+      # creating an instance of split_pdf_log
+      filename = original_filename.nil? ? File.basename(path) : original_filename
+      num_pages = pdf.pages.length
+      complete_dir = File.join(exam_template.base_path, 'complete')
+      incomplete_dir = File.join(exam_template.base_path, 'incomplete')
+      num_groups_in_complete = get_num_groups_in_dir(complete_dir)
+      num_groups_in_incomplete = get_num_groups_in_dir(incomplete_dir)
+
+      success = !partial_exams.empty?
```

Comment:
This variable is redundant. You should remove it from here and from the schema. Please do not make any schema changes without consulting me first, as you did for the exam template id.